ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -311,13 +311,43 @@ extension SignalProducerType {
 	}
 
 	/// Creates a Signal from the producer, then adds exactly one observer to
-	/// the Signal, which will invoke the given callbacks when events are
+	/// the Signal, which will invoke the given callback when `next` events are
 	/// received.
 	///
 	/// Returns a Disposable which can be used to interrupt the work associated
 	/// with the Signal, and prevent any future callbacks from being invoked.
-	public func start(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable {
-		return start(Event.sink(next: next, error: error, completed: completed, interrupted: interrupted))
+	public func startNext(next: T -> ()) -> Disposable {
+		return start(Event.sink(next: next))
+	}
+
+	/// Creates a Signal from the producer, then adds exactly one observer to
+	/// the Signal, which will invoke the given callback when an `error` event is
+	/// received.
+	///
+	/// Returns a Disposable which can be used to interrupt the work associated
+	/// with the Signal.
+	public func startError(error: E -> ()) -> Disposable {
```

Comment:
I agree the naming is weird, and I think your suggestion is better. I think it is still a bit awkward to distinguish them by forcing you to name the first parameter, but I don't have a better suggestion really :)